### ISSUE TITLE
[ENH]: CPU and GPU benchmarks

### ISSUE THREAD

**Issue by celestinoxp** [pycaret/pycaret#3055]
### Describe the feature you want to add to this project

This idea is for anyone who uses a laptop or pc from home.

I'm wondering what it will be like to use pycaret with a newer processor.

I have a Ryzen 7 4800H with 16GB of RAM (3200mhz) and honestly I don't think it satisfies me in my projects.

I don't like to wait for hours for the end of the process and in the end to have an accuracy of less than 10%, that is, I had to spend several hours to know that something is wrong with my project. Worst of all is that I may have to change only 1 line, for example in the setup and wait several hours for the project to be executed again and again see that the results are not the intended ones.

I'm curious to see how it will perform with the latest technology, such as the new AMD Ryzen 7950 or Intel 13900K.

And the DDR5 ram memories, do they make much difference?

And the graphics cards (GPU) of last generation, does it have a great performance in a catboost or lightgbm model?

It is based on these questions that I think we should create a benchmark system for some standard pycaret projects in which the objective is to see the performance of various technologies.

The idea would be for the user to benchmark and the data to be stored in a pycaret database. The data to be saved would be the dataset, algorithm used and pc data such as cpu, gpu and ram.

What do you think about this?

### Describe your proposed solution

.

**Comment by Beenjamming** [comment]
This is a great idea.  Taking this a step further, would be great to have this as a built in function that pulls the system attributes and the output time of running the sample data.  Something like this:

def system_benchmark():
    import platform
    print("Python version: {}".format(platform.python_version()))
    print("System: {}".format(platform.system()))
    print("Processor: {}".format(platform.processor()))
    print("Machine: {}".format(platform.machine()))

**Comment by moezali1** [comment]
Closing as part of the 4.0 backlog cleanup — see #4177.

PyCaret 4.0 is a ground-up rewrite (OOP-only API, sklearn-native engine, Python 3.11+, sklearn ≥ 1.7) and almost all pre-4.0 issues either reference removed APIs (`setup()`, `plot_model`, `evaluate_model`, `interpret_model`, `automl`, the legacy time-series module, the Streamlit dashboard, …) or no longer reproduce on the new code path.

**If this still affects you in 4.0, please re-file as a new issue** with:

- the experiment constructor call (`Experiment(...)`),
- the operation that fails,
- the full traceback,
- your Python / sklearn / pycaret versions.

Resources for 4.0:

- Docs: https://pycaret.org/docs
- Migrate from 3.x: https://pycaret.org/docs/guides/migrating-from-3-x
- API reference: https://pycaret.org/reference
- FAQ: https://pycaret.org/docs/resources/faqs

Thanks for filing — and apologies for the long latency. The repo had drifted; we're starting clean.
